webpack/assets/javascripts/react_app/components/HostsIndex/index.js

```diff
@@ -465,6 +465,17 @@ const HostsIndex = () => {
               selectedCount={selectedCount}
             />
           </SplitItem>
+          <SplitItem>
+            <Button
+              component="a"
+              ouiaId="export-hosts-button"
+              href={foremanUrl(`${hostsIndexUrl}.csv`)}
```

Comment:
the export button constructs a static URL. If a user filters the list (e.g., name ~ production), clicking "Export" will download all hosts because the search query is not passed to the controller.
```suggestion
// Assuming 'search' is available in the component props/hooks
// You might need to import 'useSelector' or standard URI encoding
href={foremanUrl(`${hostsIndexUrl}.csv`, { search: search })}
```

Reply:
This would change the current behavior. But I think it is a useful change.